File: LeetCode/0646. Maximum Length of Pair Chain/Solution.java
/**
 * 205 / 205 test cases passed.
 * Runtime: 8 ms
 * Memory Usage: 41.4 MB 
 */
class Solution {
    public int findLongestChain(int[][] pairs) {
        int prev = Integer.MIN_VALUE, ans = 0;
        Arrays.sort(pairs, (a, b) -> a[1] - b[1]);
        for (int[] p: pairs) {
            if (p[0] > prev) {
                prev = p[1];
                ++ ans;
            }
        }
        return ans;
    }
}

File: LeetCode/0646. Maximum Length of Pair Chain/solution.cpp
/**
 * 205 / 205 test cases passed.
 * Runtime: 44 ms
 * Memory Usage: 22 MB 
 */
class Solution {
public:
    int findLongestChain(vector<vector<int>>& pairs) {
        int prev = INT_MIN;
        int ans = 0;
        sort(pairs.begin(), pairs.end(), [](const vector<int> &a, const vector<int> &b) {
            return a[1] < b[1];
        });
        for (auto &p: pairs) {
            if (p[0] > prev) {
                prev = p[1];
                ++ ans;
            }
        }
        return ans;
    }
};

